### PR TITLE
feat(dashboards): Add drag zoom to `LineChartWidget`

### PR DESCRIPTION
**Pull request by gggritso**
**e.g.,**

https://github.com/user-attachments/assets/2fbe5c5c-72c1-49c0-bcc0-3c146c89895c

### CODE REVIEW

File: static/app/views/dashboards/widgets/lineChartWidget/lineChartWidget.stories.tsx [hunk 1]
```diff
@@ -88,3 +110,28 @@ export default storyBook(LineChartWidget, story => {
...
+        if (moment(datum.timestamp).isBefore(moment.utc(start))) {
+          return false;
+        }
+      }
+
+      if (end) {
+        if (moment(datum.timestamp).isAfter(moment.utc(end))) {
+          return false;
+        }
+      }
```
Comment (JonasBa) [inline static/app/views/dashboards/widgets/lineChartWidget/lineChartWidget.stories.tsx:132]: Is datum.timestamp already utc here?

Unless you expect other conditions, you could just shove it all into a single check.
```suggestion
        if (start && moment(datum.timestamp).isBefore(moment.utc(start))) {
          return false;
        }

        if (end && moment(datum.timestamp).isAfter(moment.utc(end))) {
          return false;
        }
```

File: static/app/views/dashboards/widgets/lineChartWidget/lineChartWidget.stories.tsx [hunk 2]
```diff
@@ -88,3 +110,28 @@ export default storyBook(LineChartWidget, story => {
 const MediumWidget = styled('div')`
   width: 420px;
 `;
+
+function filterTimeSeriesToSelection(
```
Comment (JonasBa) [inline static/app/views/dashboards/widgets/lineChartWidget/lineChartWidget.stories.tsx:114]: this does filtering + transformation to the time series :)
```suggestion
function toTimeSeriesSelection(
```

File: static/app/views/dashboards/widgets/lineChartWidget/lineChartWidget.stories.tsx [hunk 3]
```diff
@@ -88,3 +110,28 @@ export default storyBook(LineChartWidget, story => {
...
+  timeSeries: TimeseriesData,
+  start: DateString | null,
+  end: DateString | null
```
Comment (JonasBa) [inline static/app/views/dashboards/widgets/lineChartWidget/lineChartWidget.stories.tsx:117]: I will live long enough to see this type be removed from the codebase 🤣

File: static/app/views/dashboards/widgets/lineChartWidget/lineChartWidget.stories.tsx (hunk 2 above)
Comment (JonasBa) [inline static/app/views/dashboards/widgets/lineChartWidget/lineChartWidget.stories.tsx:114]: A big thank you for making this a separate function and not defining it inside the component body, this is a big part in how we make components more readable ❤️

**Review by JonasBa — APPROVED** [review]

File: static/app/views/dashboards/widgets/lineChartWidget/lineChartWidget.stories.tsx (hunk 1 above)
Comment (gggritso) [reply static/app/views/dashboards/widgets/lineChartWidget/lineChartWidget.stories.tsx]: Ah, no, `datum.timestamp` is not in UTC, it's a timestamp with a timezone like `"2024-10-24T15:00:00-04:00"`

**Pull request merged**